cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -286,4 +394,95 @@ public void finished(Throwable failureCause) {
       }
     }
   }
+
+  /**
+   * Aggregate the sizes of all stream writers. A notification is published if the aggregated
+   * size is higher than a threshold.
+   */
+  private final class Aggregator extends AbstractScheduledService {
```

Comment:
Actually, there is some difference in the two classes. The aggregation logic, the ``streamFeed``, ``streamBaseCount`` construction seems similar though. I completely understand though that the classes are private and final to two different ``StreamService`` environments. Just saw a fair bit of similar code so the mind wondered. I would say see if the refactoring is simple/makes code more maintainable, else keep it as is :)